### PR TITLE
rubocop: Fix `Cask/StanzaOrder` offenses for `on_*` blocks

### PR DESCRIPTION
**Pull request by issyl0**
Fixes new RuboCop offenses from https://github.com/Homebrew/brew/pull/14976.

----

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [ ] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [ ] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask-drivers/search?q=is%3Aclosed&type=Issues).
- [ ] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --new-cask <cask>` worked successfully.
- [ ] `brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/xerox-print-driver.rb [hunk 1]
```diff
@@ -26,6 +29,9 @@
       skip "Legacy version"
     end
   end
+  on_high_sierra :or_newer do
+    pkg "Xerox Drivers #{version.sub(/_.*/, "")}.pkg"
```
Comment (bevanjkay) [inline Casks/xerox-print-driver.rb:33]: @issyl0 This is going to cause an issue, because the `version` for mojave and newer is declared after this stanza.

Comment (bevanjkay) [reply Casks/xerox-print-driver.rb:33]: To fix it we will have to move the `pkg` into the other blocks instead of it being on its own.

Comment (issyl0) [reply Casks/xerox-print-driver.rb:33]: Ahhh yes of course.

**Comment by issyl0** [comment]
Thanks again @miccal! I hadn't had time to get around to this!

**Review by bevanjkay — APPROVED** [review]

**Pull request merged**